bindings/wasm/src/tangle/resolver.rs

```diff
@@ -35,6 +37,19 @@ use crate::tangle::WasmClient;
 #[wasm_bindgen(js_name = Resolver)]
 pub struct WasmResolver(pub(crate) Rc<Resolver<Rc<Client>>>);
 
+#[wasm_bindgen]
+extern "C" {
+  // Workaround for Typescript type annotations on async function returns.
+  #[wasm_bindgen(typescript_type = "Promise<void>")]
+  pub type PromiseVoid;
+
+  #[wasm_bindgen(typescript_type = "ResolvedDocument | null")]
+  pub type OptionResolvedDocument;
+
+  #[wasm_bindgen(typescript_type = "ResolvedDocument[] | null")]
+  pub type OptionArrayResolvedDocument;
```

Comment:
We typically use `| undefined` rather than `null` because the former allows developers to omit the parameter entirely if all subsequent parameters are also optional and omitted.

This is why `wasm-bindgen` generate `ResolvedDocument | undefined` and `napi` generates `ResolvedDocument | null | undefined` for `Option<ResolvedDocument>`. I may be wrong on this but this is what I recall.


```suggestion
  #[wasm_bindgen(typescript_type = "ResolvedDocument | undefined")]
  pub type OptionResolvedDocument;

  #[wasm_bindgen(typescript_type = "ResolvedDocument[] | undefined")]
  pub type OptionArrayResolvedDocument;
```

Reply:
I think `null` can also be omitted. This is what I do in the `create_vp.js` example or am I missing something? Regardless of how this works exactly I would be happy to keep this consistent with the rest of our code so I will change it to `undefined`.

Reply:
I think it works in the examples because they use Javascript instead of Typescript and both `null` and `undefined` correspond to `Option::None` through serde. In Typescript omitting `ResolvedDocument | null` would refuse to compile since that's technically passing `undefined`.